src/coreclr/jit/compiler.h

```diff
@@ -1036,6 +1036,13 @@ class LclVarDsc
         m_layout = layout;
     }
 
+    // Change the layout to one that may not be compatible.
+    void ChangeLayout(ClassLayout* layout)
+    {
+        assert(varTypeIsStruct(lvType));
+        m_layout = layout;
+    }
```

Comment:
Replace `GrowBlockLayout` with this?

Reply:
Not sure ... neither is widely used, both likely get inlined and evaporate in release.